### PR TITLE
Add `--no-install-recommends` to Ubuntu shared scripts

### LINKED ISSUE #762: R 4.4.0 is unintentionally installed in base images
R 4.4.0 is being unintentionally installed into base images and all other downstream images. This is likely due to `--no-install-recommends` not being used in the shared installer scripts which cascades into `r-recommended` getting installed as a recommended package from `r-base-core`.

### PR DESCRIPTION
**Pull request by ianpittwood**
Fixes #762 

- Prevent `r-recommended` from being unintentionally installed
- Stop installing `r-base-core` since it adds an R installation unnecessarily
- Check that `r-recommended` and system R version are not present
- Add `--no-install-recommends` to all shared scripts, it's best practice for images

### CODE REVIEW

File: product/base/scripts/ubuntu/install_python.sh [hunk 1]
```diff
@@ -73,7 +73,7 @@ fi
 PYTHON_BIN="${PREFIX}/${PYTHON_VERSION}/bin/python"
 
 # Set apt options
-APT_ARGS="-o DPkg::Lock::Timeout=60 -y -qq"
+APT_ARGS="-o DPkg::Lock::Timeout=60 -y -qq --no-install-recommends"
```
Comment (bschwedler) [inline product/base/scripts/ubuntu/install_python.sh:76]: Do we need the second definition?

File: product/base/scripts/ubuntu/install_r.sh [hunk 2]
```diff
@@ -85,7 +85,7 @@ fi
 R_BIN="${PREFIX}/${R_VERSION}/bin/R"
 
 # Set apt options
-APT_ARGS="-o DPkg::Lock::Timeout=60 -y -qq"
+APT_ARGS="-o DPkg::Lock::Timeout=60 -y -qq --no-install-recommends"
```
Comment (bschwedler) [inline product/base/scripts/ubuntu/install_r.sh:88]: Do we need the second definition?

File: product/base/test/goss.yaml [hunk 3]
```diff
@@ -45,6 +47,8 @@ file:
     exists: true
   /opt/python/{{.Env.PYTHON_VERSION_ALT}}/bin/python3:
     exists: true
+  /usr/lib/R/bin/R:
+    exists: false
```
Comment (bschwedler) [inline product/base/test/goss.yaml:51]: Nice!

File: product/base/scripts/ubuntu/install_python.sh (hunk 1 above)
Comment (ianpittwood) [reply product/base/scripts/ubuntu/install_python.sh:75]: Fair point, removed

File: product/base/scripts/ubuntu/install_r.sh (hunk 2 above)
Comment (ianpittwood) [reply product/base/scripts/ubuntu/install_r.sh:87]: Removed

File: product/base/test/goss.yaml [hunk 4]
```diff
@@ -6,6 +6,8 @@ package:
     installed: true
   cracklib-runtime:
     installed: true
+  r-recommended:
+    installed: false
```
Comment (GCRev) [inline product/base/test/goss.yaml:10]: I'm assuming this performs the same work that `install_r_dependencies` did?

**Review by GCRev — APPROVED** [review]
LGTM

File: product/base/test/goss.yaml (hunk 4 above)
Comment (ianpittwood) [reply product/base/test/goss.yaml:10]: No this is just checking that no system package version of R is getting installed. All of the dependencies should now be in `ubuntu2204_packages.txt`.

**Pull request merged**